test/240_dns_add_name_test.sh

```diff
@@ -19,11 +20,17 @@ weave_on $HOST1 dns-add $C2 c2 -h $NAME2
 assert_dns_record $HOST1 c1 $NAME2 $C2
 
 weave_on $HOST1 dns-add $C1 c1 -h $NAME1
+weave_on $HOST1 dns-add $C1 c1 -h $NAME3
 
-assert "exec_on $HOST1 c1 getent hosts $C1 | tr -s ' '" "$C1 $NAME1"
+assert "exec_on $HOST1 c1 getent hosts $NAME1 | tr -s ' '" "$C1 $NAME1"
+assert "exec_on $HOST1 c1 getent hosts $NAME3 | tr -s ' '" "$C1 $NAME3"
 
-weave_on $HOST1 dns-remove $C1 c1
+weave_on $HOST1 dns-remove $C1 c1 -h $NAME1
+
+whitely echo "Waiting 30s for cache expiry..."
+sleep 30
```

Comment:
I mentioned this possibility in my initial comments - I think we can change it to use -no-cache once #615 is merged...